modules/core/src/main/java/org/apache/ignite/internal/processors/cache/distributed/dht/preloader/GridDhtPartitionsExchangeFuture.java

```diff
@@ -4150,7 +4152,7 @@ private void validatePartitionsState() {
     /**
      * @param resetOwners True if reset partitions state needed, false otherwise.
      */
-    private void assignPartitionsStates(boolean resetOwners) {
+    private void assignPartitionsStates(boolean resetOwners, boolean resetOwnersForStartedCacheGroupsOnly) {
```

Comment:
This change makes logic of assignPartitionsStates more complex. For some reason, it should be aware which caches have started during this exchange.
Can we pass groupIds on which we should reset owners instead of two boolean flags? We can treat null set as "reset owners for all groups" for simplicity.